src/src/tx_main.cpp

```diff
@@ -209,20 +211,25 @@ void ICACHE_RAM_ATTR Generate4ChannelData_11bit()
 #endif
 }
 
-void ICACHE_RAM_ATTR GenerateSwitchChannelData()
+void ICACHE_RAM_ATTR GenerateMSPData()
 {
   uint8_t PacketHeaderAddr;
-  PacketHeaderAddr = (DeviceAddr << 2) + SWITCH_DATA_PACKET;
+  PacketHeaderAddr = (DeviceAddr << 2) + MSP_DATA_PACKET;
   Radio.TXdataBuffer[0] = PacketHeaderAddr;
-  Radio.TXdataBuffer[1] = ((CRSF_to_UINT10(crsf.ChannelDataIn[4]) & 0b1110000000) >> 2) +
-                          ((CRSF_to_UINT10(crsf.ChannelDataIn[5]) & 0b1110000000) >> 5) +
-                          ((CRSF_to_UINT10(crsf.ChannelDataIn[6]) & 0b1100000000) >> 8);
-  Radio.TXdataBuffer[2] = (CRSF_to_UINT10(crsf.ChannelDataIn[6]) & 0b0010000000) +
-                          ((CRSF_to_UINT10(crsf.ChannelDataIn[7]) & 0b1110000000) >> 3);
-  Radio.TXdataBuffer[3] = Radio.TXdataBuffer[1];
-  Radio.TXdataBuffer[4] = Radio.TXdataBuffer[2];
-  Radio.TXdataBuffer[5] = Radio.NonceTX;
-  Radio.TXdataBuffer[6] = FHSSgetCurrIndex();
+  Radio.TXdataBuffer[1] = MSPPacket.function;
+  Radio.TXdataBuffer[2] = MSPPacket.payloadSize;
+  Radio.TXdataBuffer[3] = 0;
+  Radio.TXdataBuffer[4] = 0;
+  Radio.TXdataBuffer[5] = 0;
+  Radio.TXdataBuffer[6] = 0;
+  if (MSPPacket.payloadSize <= 4)
```

Comment:
do we want a debug output on the 'else' side of this if statement?

Reply:
Added